scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -2093,7 +2093,34 @@ class Router(formatOps: FormatOps) {
           if (close.isEmpty) Split.ignored else Split(Space, 1)
 
         case Newlines.classic if style.align.arrowEnumeratorGenerator =>
-          Split(Space, 1).withIndent(StateColumn, expire, After)
+          Split(Space, 1)
+            .withIndent(StateColumn, expire, After)
+            /**
+              * This gap is necessary for pretty alignment multiline expressions
+              * on the right-side of enumerator.
+              * Without:
+              * ```
+              * for {
+              *    a <- new Integer {
+              *          value = 1
+              *        }
+              *   x <- if (variable) doSomething
+              *       else doAnything
+              * }
+              * ```
+              *
+              * With:
+              * ```
+              * for {
+              *    a <- new Integer {
+              *           value = 1
+              *         }
+              *   x <- if (variable) doSomething
+              *        else doAnything
+              * }
+              * ```
+              * */
+            .withIndent(Num(1), expire, After)
```

Comment:
is it because `<-` is two-characters long? would it still work with the single-character unicode left arrow?

Reply:
No, it's because of space after arrow. Added test for unicode